internal/gensupport/resumable.go

```diff
@@ -241,15 +253,23 @@ func (rx *ResumableUpload) Upload(ctx context.Context) (resp *http.Response, err
 			default:
 			}
 
-			resp, err = rx.transferChunk(ctx)
+			rCtx, cancel := context.WithTimeout(ctx, transferTimeout)
+			defer cancel()
+
+			resp, err = rx.transferChunk(rCtx)
 
 			var status int
 			if resp != nil {
 				status = resp.StatusCode
 			}
 
+			if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
How do we distinguish between a timeout in ctx vs rCtx?

I think we'd only want to retry if we get deadline exceeded from rCtx, but ctx is still good.

Reply:
I feel below condition is enough to check this `I think we'd only want to retry if we get deadline exceeded from rCtx, but ctx is still good.` Removed that condition.